cupyx/scipy/sparse/coo.py

```diff
@@ -215,7 +216,7 @@ def sum_duplicates(self):
         if self.data.size == 0:
             self._has_canonical_format = True
             return
-        keys = cupy.stack([self.col, self.row])
+        keys = cupy.stack([self.row, self.col])
```

Comment:
Are you sure about this change? The description you quoted states:
"the index arrays are first sorted by row indices and then within the same row by column indices"

Since lexsort uses the last row as the primary sort key, didn't the already existing code match that description?

In either case, it seems it can be verified empirically against SciPy. If all test cases pass either way it seems that would indicate a hole in our test coverage!

Reply:
I need your help, Gregory. I've been looking into this for a few hours, and I am super confused now. 

It seems the change at this line neither affects speed nor correctness, at least with my test script. (And FYI, the whole test suite `tests/cupyx_tests/scipy_tests/sparse_tests/` passed either way!) The only thing that matters (for speed) is to set `_has_canonical_format=True`, which makes sense because when calling `sum_duplicates()` both CSC & CSR converts to COO (that's why changes in `coo.py` affects CSR), and this flag would provide a fast path. Does it mean `keys`, `lexsort`, etc here are simply dead code? I am lost...

Reply:
I haven't looked at it closely, but was just wandering what motivated the change. Ca n try to take a look tomorrow

Reply:
I now incline that this was simply an overlook of previous PR and that we don't need to change this line. After `coo.sum_duplicates()` and `cusparse.coosort()`, `_has_canonical_format` should always be `True`. The only concern is that `cusparse.coosort()` sorts rows by default, and for `tocsc()` it doesn't sound right, so I will address both issues.

Reply:
The csc format is stored in column-major format, whereas csr is in row-major, 
so I think a CSC matrix is stored internally as a transpose of a CSR matrix.

Reply:
CSR/CSC matrix internally stores 3 arrays: `data` of size `nnz`, `indices` of size `nnz`, and `indptr` of size `m+1`/`n+1`. This is the same for both cuSPASE and SciPy. A transpose operation would need to update indices` and `indptr`.

Reply:
Revisiting this line, interestingly I found SciPy actually does what @smarkesini did (sort by col first):
https://github.com/scipy/scipy/blob/adc4f4f7bab120ccfab9383aba272954a0a12fb0/scipy/sparse/coo.py#L543. I wonder why no test suite has caught this deviation so far (and how to write one)...